cli/src/main/java/denominator/cli/Denominator.java

```diff
@@ -426,6 +427,44 @@ void overrideFromEnv(Map<String, String> env) {
       }
     }
 
+    static void setProxyFromEnv() {
+      String envHttpProxy = System.getenv("HTTP_PROXY");
```

Comment:
this looks refactorable.. I think CLI has guava (implicitly), so you can use default parsing in HostAndPort. Next, I think it is worth parameterizing the 's' so that the logic isn't completely copy/paste.

ex.
```java
setProxyHostAndPort("http", System.getenv("HTTP_PROXY"));
setProxyHostAndPort("https", System.getenv("HTTPS_PROXY"));
```